packages/shared_preferences/shared_preferences_android/pigeons/messages.dart

```diff
@@ -0,0 +1,35 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:pigeon/pigeon.dart';
+
+@ConfigurePigeon(PigeonOptions(
+  input: 'pigeons/messages.dart',
+  javaOut:
+      'android/src/main/java/io/flutter/plugins/sharedpreferences/Messages.java',
+  javaOptions: JavaOptions(
+      className: 'Messages', package: 'io.flutter.plugins.sharedpreferences'),
+  dartOut: 'lib/messages.g.dart',
+  dartTestOut: 'test/messages_test.g.dart',
+  copyrightHeader: 'pigeons/copyright.txt',
+))
+@HostApi(dartHostTestHandler: 'TestSharedPreferencesApi')
+abstract class SharedPreferencesApi {
+  @TaskQueue(type: TaskQueueType.serialBackgroundThread)
+  bool? remove(String key);
```

Comment:
Why are all of these return values nullable? What does a null return indicate that is different from `true` or `false`?

Reply:
Yes, I have kept them nullable in case some error or exception occurs, so we have the safety mechanism to handle that just in case. If you want me to change it, to non-nullable I can do that as well.

Reply:
> Yes, I have kept them nullable in case some error or exception occurs, so we have the safety mechanism to handle that just in case.

I don't follow, since exceptions don't return a value. Can you describe a scenario in which null is returned?

Reply:
Oh yeah, if it succeeds then it's true otherwise it is false for all other cases. In case of exception, the execution is stopped so there is no return in the first place. That makes sense, I will change it to non-nullable